task8/ServletsOnMaven/src/main/webapp/resources/js/View.js

```diff
@@ -602,29 +339,31 @@ Input the password
 		loginWindowOKButton.className = "button_OK";
 		loginWindowOKButton.setAttribute("type", "submit");
 		loginWindowOKButton.setAttribute("value", "OK");
-		loginWindowOKButton.addEventListener("click", ()=>{
+		loginWindowOKButton.addEventListener("click", async () => {
 			let fields = this.loginWindow.getElementsByTagName("input");
 			let login = fields[0].value;
 			let password = fields[1].value;
-			let user = users.get(login);
-			if (user === undefined) {
-				this.viewErrorWindow("No such login. Try again");
-				fields[0].value = "";
-				fields[1].value = "";
+			if (login === "") {
+				this.viewErrorWindow("The login or password field is empty. Try again", function () {
+					view.viewWindowToLogin();
+				});
+				fields[0].value = fields[1].value = "";
 				return;
 			}
-			else if (user.password !== password) {
-				this.viewErrorWindow("Incorrect password. Try again");
-				fields[0].value = "";
-				fields[1].value = "";
+			let realPassword = "";
+			await sendGetRequest("/users", "username", login).then(data => realPassword = data);
```

Comment:
Sending password to the client is too unsecure way, that comprising (not even that, but hashed values should be compared, but nvm) must be on the server side